vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -1264,7 +1265,7 @@ protected Grid<T> getGrid() {
      */
     private class DetailsManager extends AbstractGridExtension<T> {
 
-        private final HashSet<T> detailsVisible = new HashSet<>();
+        private final HashMap<Object, T> detailsVisible = new LinkedHashMap<>();
```

Comment:
The order of items doesn't seem to be important, so perhaps HashMap would be more appropriate?

```suggestion
        private final HashMap<Object, T> detailsVisible = new HashMap<>();
```

Reply:
Probably best to keep the existing behavior, for example, someone might have tests that rely on the order in which details were opened.

Reply:
Well, previously, a HashSet was used, which doesn't guarantee order. However, the selected items, which I used for inspiration, are stored in a LinkedHashMap, which preserves insertion order.